packages/salesforcedx-apex-replay-debugger/src/core/heapDump.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ApexExecutionOverlayResultCommandSuccess } from '../commands';
+
+export class ApexHeapDump {
+  private readonly heapDumpId: string;
+  private readonly className: string;
+  private readonly namespace: string;
+  private readonly line: number;
+  private overlaySuccessResut:
+    | ApexExecutionOverlayResultCommandSuccess
+    | undefined;
+  public constructor(
+    heapDumpId: string,
+    className: string,
+    namespace: string,
+    line: number
+  ) {
+    this.heapDumpId = heapDumpId;
+    this.className = className;
+    this.namespace = namespace;
+    this.line = line;
+  }
+  public getHeapDumpId(): string {
+    return this.heapDumpId;
+  }
+  public getClassName(): string {
+    return this.className;
+  }
+  public getNamespace(): string {
+    return this.namespace;
+  }
+  public getLine(): number {
+    return this.line;
+  }
+  public getOverlaySuccessResult():
+    | ApexExecutionOverlayResultCommandSuccess
+    | undefined {
+    return this.overlaySuccessResut;
+  }
+  public setOverlaySuccessResult(
+    overlaySuccessResult: ApexExecutionOverlayResultCommandSuccess
+  ): void {
+    this.overlaySuccessResut = overlaySuccessResult;
+  }
+  /* tslint:disable */
+  public toString = (): string => {
```

Comment:
I thought I left a comment here but I must not have clicked "Add Single Comment".

Why do we need to write the method using the `public toString = (): string => {..}` style? Would a normal override work? That would seem to eliminate the issues that you are seeing with tslint.

See https://stackoverflow.com/questions/35361482/typescript-override-tostring/45320960 and the fiddle at https://jsfiddle.net/sy8wttvw/

Reply:
@vazexqi 
public toString(): string { *versus* public toString = (): string => {
The basic difference between the two is that the arrow function doesn't bind their own this, so this is what you'd expect. It's actually discussed in the first answer of the link you posted.

Reply:
Does yours produce a different answer if you use `public toString(): string`?

I have a fiddle here where I am using `public toString(): string` and I don't see any difference. https://jsfiddle.net/sy8wttvw/55/

I'm trying to understand why we need to use that style in our use case.

Reply:
@vazexqi, I changed the function it's no longer an arrow function.

Here's a good read on arrow functions as to why I'd done what I'd done. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
I _think_ I understand arrow functions and when to use them. There are two things here so I don't want to conflate them:

1. Why are we using an arrow function _here_? It stood out to me not because of the tslint but because it's a different format for defining the method. That's what prompted my question to help me understand why we need that. It seems that we don't need it (no loss of functionality, or malfunction) so the normal function declaration is better.
1. The tslint issue is true and could be annoying. It means that there is a mismatch between prettier and tslint. This one should be fixed. I don't know off the top of my head where it is. It might be good to just try it in a simple context and start disabling prettier and see what happens. Then we can trace down the exact tslint rule. That's how I would do it if I were in this situation. I won't be able to tell you the rule without doing that process myself either.